core/src/main/java/org/jenkins/ui/icon/IconSet.java

```diff
@@ -92,10 +92,14 @@ public static String getSymbol(String name, String title, String tooltip, String
             String symbol = symbolsForLookup.get(translatedName);
             symbol = symbol.replaceAll("(class=\")[^&]*?(\")", "$1$2");
             symbol = symbol.replaceAll("(tooltip=\")[^&]*?(\")", "");
+            symbol = symbol.replaceAll("(html-tooltip=\")[^&]*?(\")", "");
             symbol = symbol.replaceAll("(id=\")[^&]*?(\")", "");
-            if (!tooltip.isEmpty()) {
+            if (!tooltip.isEmpty() && htmlTooltip.isEmpty()) {
                 symbol = symbol.replaceAll("<svg", "<svg tooltip=\"" + Functions.htmlAttributeEscape(tooltip) + "\"");
             }
+            if (!htmlTooltip.isEmpty()) {
+                symbol = symbol.replaceAll("<svg", "<svg html-tooltip=\"" + htmlTooltip + "\"");
```

Comment:
the point of this attribute is to take HTML, escaping it would break it?

It's documented that you must escape any user supplied content before passing it on

Reply:
String concatenation without escaping negates one level of escaping. So escaping once it the caller, as documentation suggests, won't be enough when working with user-provided content.

I've tested with the following scenario:
1. In [BuildButtonColumn/column.jelly#L44](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/hudson/views/BuildButtonColumn/column.jelly#L44) add `htmlTooltip="${h.htmlAttributeEscape(title)}"` as `l:icon` attribute. Escaping it once as documentation suggests.
2. Create a job and set a display name for it to e.g. `123 <b>456</b> <img/src/onerror=alert(1)>`.
3. Come back to the dashboard.

Reply:
Thanks I tested it out and it does work as required.